src/main/scala/nl/knaw/dans/lib/logging/DebugEnhancedLogging.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib.logging
+
+import com.typesafe.scalalogging.LazyLogging
+
+trait DebugEnhancedLogging extends LazyLogging {
+  /**
+   * Logs a trace level statement with the function name, parameter names and parameter values.
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    trace(s, i)
+   *    // Your code ...
+   *   }
+   *
+   *   // Function call
+   *   someFunction("My input string", 2)
+   *
+   *   // This will result in a logging line like this (if the TRACE log level is enabled):
+   *   [TRACE] someFunction [(s, i)]:  ("My input string",2)
+   *
+   * }}}
+   *
+   * @param value list of parameter values
+   * @param enclosing implicitly passed in context from the sourcecode library
+   * @tparam V type parameter, no idea how it works
+   */
+  def trace[V](value: sourcecode.Text[V])(implicit enclosing: sourcecode.Name) = {
+    logger.trace(s"${enclosing.value} [${value.source}]: ${value.value}")
+  }
+
+
+  /**
+   * Logs a custom debug level statement with the calling function name prefixed
+   *
+   * Example:
+   * {{{
+   *   // Function definition
+   *   def someFunction(s: String, i: Int) = {
+   *    // Your code ...
+   *    debug("Doing important stuff here")
+   *   }
+   *
+   *
+   *   // This will result in a logging line like this (if the DEBUG log level is enabled):
+   *   [DEBUG] someFunction: Doing important stuff here
+   *
+   * }}}
+   * @param s
+   * @param enclosing
+   */
+  def debug(s: String)(implicit enclosing: sourcecode.Name) = logger.debug(s"${enclosing.value}: $s")
```

Comment:
The rationale for this `debug` method is that I would like a short prefix to a normal debug-message that gives a clue about the context, not a completely specified one. I don't see (yet) how I can achieve that with the logback caller construct. Also, it should **not** impact the other logging levels. Anyway, if we figure out how to do that in logback, that would indeed be the preferred choice. I'll leave it in for now.

About not using an explicit member, as in `logger.info("blah")`. I agree that having two different ways may be confusing. However, I'd rather use the `info("blah")` variant. We can achieve that by importing `logger._`. I won't include that in `DebugEnhancedLogging` yet, as I cannot oversee whether you'd want to be able to **not** do that, sometimes. I'll try it first in `easy-bag-store`.